tools/test-proxy/CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+  - `"x-record-mode": "record"`
+  - `"x-recording-id": "<recording-id>"`
+- Test Proxy redirects incoming request regardless of the route "magically"
+- Client POSTS to /Record/Stop, recording is written to disk.
+
+This is configured in `Startup.cs` function `MapRecording`.
+
+```cs
+foreach (var controller in new[] { "playback", "record" })
+{
+    app.MapWhen(
+        context =>
+            controller.Equals(
+                GetRecordingMode(context),
+                StringComparison.OrdinalIgnoreCase),
+        app =>
+        {
+            app.UseRouting();
+            app.UseEndpoints(
+                endpoints => endpoints.MapFallbackToController(
+                    "{*path}", "HandleRequest", controller));
+        });
+}
+```
+
+Effectively this means:
+
+> If the recording mode is record, send this to the `HandleRequest` function in the `Record` controller. Same is true for `playback` and `Playback` controller.
+
+The `x-recording-id` header is then utilized to associate the request/response pair with the appropriate recording!
+
+## Testing
+
+As mentioned in the table before, the tests for this project reside within `Azure.Sdk.Tools.TestProxy.Tests`. When contributing a change, please:
+
+1. Run the unit tests locally to ensure nothing has been affected.
+2. Verify your change with at least one additional unit test. More is better!
+
+### Note about integration tests
+
+The tests marked with trait `Integration` are gated on access to a private github repo that is used for the purpose. Feel free to replace the target repo `Azure/azure-sdk-assets-integration` with your _own_ repo, as that will absolutely work. However that could be quite painful with the number of tests.
+
+## Gotchas
+
+When opening the project in VS2022, you may see the following:
+
+![image](https://user-images.githubusercontent.com/45376673/218187142-040881c7-2dfa-4f9f-84c6-c2058d7c878c.png)
+
+
+This is due to the fact that the `test-proxy` is part of a greater tooling repo, and must align with a pinned .NET version. To resolve this issue easily, please do the following.
+
+```bash
+dotnet --list-sdks
+3.1.426 [C:\Program Files\dotnet\sdk]
+5.0.415 [C:\Program Files\dotnet\sdk]
+6.0.113 [C:\Program Files\dotnet\sdk]
+6.0.308 [C:\Program Files\dotnet\sdk]
+6.0.401 [C:\Program Files\dotnet\sdk]
+6.0.405 [C:\Program Files\dotnet\sdk]
+7.0.101 [C:\Program Files\dotnet\sdk]
+```
+
+Find a version starting with a `6`, if you don't have one on your system, [install it!](https://dotnet.microsoft.com/en-us/download/dotnet/6.0)
+
+Once you have that, find the `global.json` at root of this repo.
+
+```json
+{
+  "msbuild-sdks": {
+    "Microsoft.Build.Traversal": "3.2.0"
+  },
+  "sdk": {
+    "version": "7.0.102",
+    "rollForward": "feature"
+  }
+}
+```
+
+Change the `version` to one that shows up in the list when you call `dotnet --list-sdks`.
```

Comment:
We're trying to configure this such that you only need to make sure to have the targeted sdk/runtime installed for the csproj, and global.json will always point to the newest version of the tooling. As such, this last step may not be needed FYI @konrad-jamrozik

Reply:
To elaborate on what @benbp said, we settled on installing both v. 6 and 7 via `install-dotnet.yml` with the idea that everything is built with v. 7 from global.json (7 as in "currently the latest") but runs on the runtime it targets. (#5405).

So there are few ways to solve it: 
- coerce everything to .net 6 as @scbedd is proposing;
- keep global.json unchanged, don't install .net 6, and ".net roll forward" as in https://github.com/Azure/azure-sdk-tools/pull/5379;
- keep global.json unchanged and install .net 6 sdk or runtime. But not sure if VS will be happy with that. It might still prioritize .net 7 coming from global.json and complain one is trying to build project targeting 6 with .net 7 sdk.

Reply:
Per my understanding `global.json` essentially wins.

Reply:
@scbedd global.json wins for the dotnet cli/tooling version required, but if you just add net 6 in the csproj, everything will still work provided you have the net 6 sdk+runtime installed.

If I'm understanding correctly, our plan is to always have the latest version of tooling, but pin our tools to the LTS, which is 6.

@konrad-jamrozik all this said, I've also had trouble making sure my local installation works seamlessly, so we should have some docs here around how to set this up because it can get pretty confusing.

Reply:
Yeah. I'll update the doc to >= .NET 6, but take action in another PR to add .NET 7 support too.